src/Ocelot.Provider.Polly/PollyQoSResiliencePipelineProvider.cs

```diff
@@ -0,0 +1,76 @@
+using Ocelot.Configuration;
+using Ocelot.Logging;
+using Ocelot.Provider.Polly.Interfaces;
+using Polly.CircuitBreaker;
+using Polly.Registry;
+using Polly.Timeout;
+
+namespace Ocelot.Provider.Polly;
+
+public class PollyQoSResiliencePipelineProvider : PollyQoSProviderBase, IPollyQoSResiliencePipelineProvider<HttpResponseMessage>
+{
+    private readonly ResiliencePipelineRegistry<OcelotResiliencePipelineKey> _resiliencePipelineRegistry;
+    private readonly IOcelotLogger _logger;
+
+    public PollyQoSResiliencePipelineProvider(IOcelotLoggerFactory loggerFactory, 
+        ResiliencePipelineRegistry<OcelotResiliencePipelineKey> resiliencePipelineRegistry)
+    {
+        _resiliencePipelineRegistry = resiliencePipelineRegistry;
+        _logger = loggerFactory.CreateLogger<PollyQoSResiliencePipelineProvider>();
+    }
+
+    public ResiliencePipeline<HttpResponseMessage> GetResiliencePipeline(DownstreamRoute route)
+    {
+        var options = route.QosOptions;
+
+        // Check if we need pipeline at all before calling GetOrAddPipeline
+        if (options is null ||
+            (options.ExceptionsAllowedBeforeBreaking == 0 && options.TimeoutValue is int.MaxValue))
+        {
+            return null; // shortcut > no qos
+        }
+
+        var currentRouteName = GetRouteName(route);
+        return _resiliencePipelineRegistry.GetOrAddPipeline<HttpResponseMessage>(
+            key: new OcelotResiliencePipelineKey(currentRouteName), 
+            configure: (builder) => PollyResiliencePipelineWrapperFactory(builder, route));
```

Comment:
(builder) =>, the parentheses are not needed

Reply:
Minor issue which can be skipped!
Imagine anonymous delegate of `Action` (without arguments). How do you define it? 😄
In case of anonim-delegate aka `() => { ... }` it is impossible to remove the parentheses! 😉 
For the case of 2+ args it is true also.
For the case of 1 arg, we could agree on usage both definitions in the code considering both of them as not issue at all 👉 
- `arg => {}`
- `(arg) => {}`

Reply:
Or we can use this anonymous delegate definition for methods without arguments and the parentheses 😉 
```c#
delegate { ... }
```
with one argument
```c#
delegate (builder) { PollyResiliencePipelineWrapperFactory(builder, route)); }
```
Do you like it? But sorry probably you don't like it!

Reply:
Ok